File: Problems/ICPC Asyut Sheets/Sheet #2 (Loops)/U. Some Sums.cpp
#include <bits/stdc++.h>
using namespace std;

int sumOfDigits(int n)
{
    int sum = 0;
    while (n)
    {
        sum += n % 10;
        n /= 10;
    }
    return sum;
}
int main()
{
    int n, a, b;
    cin >> n >> a >> b;
    int sum = 0;
    for (int i = 1; i <= n; i++)
    {
        if (sumOfDigits(i) >= a && sumOfDigits(i) <= b)
            sum += i;
    }
    cout << sum;
}

File: Problems/ICPC Asyut Sheets/Sheet #2 (Loops)/U. Some Sums.py
def sumOfDigits(n):
    sum=0
    while n>0:
        sum+=n%10
        n//=10
    return sum



n,a,b=map(int,input().split())
sum=0

for i in range(1,n+1):
    if(sumOfDigits(i)>=a and sumOfDigits(i)<=b):
        sum+=i

print(sum)